packages/flutter/test/widgets/draggable_scrollable_sheet_test.dart

```diff
@@ -1746,4 +1746,30 @@ void main() {
       areCreateAndDispose,
     );
   });
+
+  testWidgets('DraggableScrollableSheet respects shouldCloseOnMinExtent', (WidgetTester tester) async {
+    final DraggableScrollableController controller = DraggableScrollableController();
+    DraggableScrollableNotification? receivedNotification;
+
+    Future<void> pumpWidgetAndFling() async {
+      await tester.pumpWidget(boilerplateWidget(
+        null,
+        controller: controller,
+        shouldCloseOnMinExtent: false,
+        onDraggableScrollableNotification: (DraggableScrollableNotification notification) {
+          receivedNotification = notification;
+          return false;
+        },
+      ));
+
+      await tester.flingFrom(const Offset(0, 325), const Offset(0, -325), 200);
+      await tester.pumpAndSettle();
+    }
+
+    await pumpWidgetAndFling();
+    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);
+    controller.jumpTo(0.5);
+    await pumpWidgetAndFling();
+    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);
```

Comment:
A couple of thoughts relating to this test:
- Generally, `tester.pumpWidget()` only needs to be called more than once if the widget is rebuilding with a different configuration.
- It might be nice to null out `receivedNotification` to make sure the notification isn't just a leftover value from the previous fling.

```suggestion
    await tester.pumpWidget(boilerplateWidget(
      null,
      controller: controller,
      shouldCloseOnMinExtent: false,
      onDraggableScrollableNotification: (DraggableScrollableNotification notification) {
        receivedNotification = notification;
        return false;
      },
    ));

    Future<void> fling() async {
      await tester.flingFrom(const Offset(0, 325), const Offset(0, -325), 200);
      await tester.pumpAndSettle();
    }

    await fling();
    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);

    receivedNotification = null;
    controller.jumpTo(0.5);

    await fling();
    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);
```

Reply:
The test is designed to call didUpdateWidget, and currently _replaceExtent function is always called in didUpdateWidget,  there is no check for equality between the old and new widget inside. My test verifies that shouldCloseOnMinExtent is correctly assigned within _replaceExtent

Reply:
Different controllers can be used so that if didUpdateWidget is modified in the future to compare with the previous widget, the function will still be called cause the controller is new. Here example
```
DraggableScrollableController controller = DraggableScrollableController();
    DraggableScrollableNotification? receivedNotification;

    Future<void> pumpWidgetAndFling() async {
      await tester.pumpWidget(boilerplateWidget(
        null,
        controller: controller,
        shouldCloseOnMinExtent: false,
        onDraggableScrollableNotification: (DraggableScrollableNotification notification) {
          receivedNotification = notification;
          return false;
        },
      ));

      await tester.flingFrom(const Offset(0, 325), const Offset(0, -325), 200);
      await tester.pumpAndSettle();
    }

    await pumpWidgetAndFling();
    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);
    controller.jumpTo(0.5);
    
    controller = DraggableScrollableController();
    await pumpWidgetAndFling();
    expect(receivedNotification!.shouldCloseOnMinExtent, isFalse);
```